aptos-move/block-executor/src/view.rs

```diff
@@ -89,6 +95,267 @@ pub(crate) struct ParallelState<'a, T: Transaction, X: Executable> {
     captured_reads: RefCell<CapturedReads<T>>,
 }
 
+fn get_delayed_field_value_impl<T: Transaction>(
+    captured_reads: &RefCell<CapturedReads<T>>,
+    versioned_delayed_fields: &dyn TVersionedDelayedFieldView<T::Identifier>,
+    wait_for: &dyn TWaitForDependency,
+    id: &T::Identifier,
+    txn_idx: TxnIndex,
+) -> Result<DelayedFieldValue, PanicOr<DelayedFieldsSpeculativeError>> {
+    // We expect only DelayedFieldReadKind::Value (which is set from this function),
+    // to be a "full materialized/aggregated" read, and so we don't use the value
+    // from HistoryBounded reads.
+    // If we wanted to make it more dynamic, we could have a type of the read value
+    // inside HistoryBounded
+    let delayed_read = captured_reads
+        .borrow()
+        .get_delayed_field_by_kind(id, DelayedFieldReadKind::Value);
+    if let Some(data) = delayed_read {
+        if let DelayedFieldRead::Value { value, .. } = data {
+            return Ok(value);
+        } else {
+            let err =
+                code_invariant_error("Value DelayedField read returned non-value result").into();
+            captured_reads
+                .borrow_mut()
+                .capture_delayed_field_read_error(&err);
+            return Err(err);
+        }
+    }
+
+    loop {
+        match versioned_delayed_fields.read(id, txn_idx) {
+            Ok(value) => {
+                captured_reads.borrow_mut().capture_delayed_field_read(
+                    *id,
+                    false,
```

Comment:
So, if `update` flag is false, create a new captured read, and if `update` flag is true, update an existing captured read?

Reply:
yes. this is mostly for checks inside of CapturedReads on what to accept. 

maybe "update_to_same_kind" is better name. or maybe I should infer it inside capture_delayed_field_read, hm